drivers/weather/weatherflow.cpp

```diff
@@ -0,0 +1,513 @@
+  file called LICENSE.
+*******************************************************************************/
+
+#include "weatherflow.h"
+
+#include "locale_compat.h"
+
+#include <cstring>
+#include <sstream>
+#include <iomanip>
+#include <limits>
+#include <thread>
+
+// We declare an auto pointer to WeatherFlow.
+std::unique_ptr<WeatherFlow> weatherFlow(new WeatherFlow());
+
+static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string *)userp)->append((char *)contents, size * nmemb);
+    return size * nmemb;
+}
+
+WeatherFlow::WeatherFlow()
+{
+    setVersion(1, 0);
+    setWeatherConnection(CONNECTION_NONE);
+
+    // Initialize CURL
+    curl_global_init(CURL_GLOBAL_DEFAULT);
+
+    // Initialize rate limiting
+    m_lastRequestTime = std::chrono::system_clock::now();
+    m_requestCount = 0;
+}
+
+WeatherFlow::~WeatherFlow()
+{
+    curl_global_cleanup();
+}
+
+const char *WeatherFlow::getDefaultName()
+{
+    return "WeatherFlow";
+}
+
+void WeatherFlow::ISGetProperties(const char *dev)
+{
+    INDI::Weather::ISGetProperties(dev);
+    defineProperty(wfAPIKeyTP);
+    defineProperty(wfStationIDTP);
+    defineProperty(wfSettingsNP);
+}
+
+bool WeatherFlow::Connect()
+{
+    if (wfAPIKeyTP[0].isEmpty())
+    {
+        LOG_ERROR("WeatherFlow API Key is not available. Please register your API key at "
+                  "https://tempestwx.com/settings/tokens and save it under Options.");
+        return false;
+    }
+
+    // Check rate limiting
+    auto now = std::chrono::system_clock::now();
+    auto timeSinceLastRequest = std::chrono::duration_cast<std::chrono::seconds>(now - m_lastRequestTime).count();
+
+    if (timeSinceLastRequest >= RATE_LIMIT_PERIOD)
+    {
+        m_requestCount = 0;
+        m_lastRequestTime = now;
+    }
+
+    if (m_requestCount >= RATE_LIMIT_REQUESTS)
+    {
+        LOG_ERROR("API rate limit exceeded. Please wait before making more requests.");
+        return false;
+    }
+
+    // Fetch station information if station ID is not provided
+    if (wfStationIDTP[0].isEmpty())
+    {
+        if (!fetchStationInfo())
+        {
+            LOG_ERROR("Failed to fetch station information. Please provide a station ID manually.");
+            return false;
+        }
+    }
+    else
+    {
+        m_stationID = wfStationIDTP[0].getText();
+    }
+
+    m_isConnected = true;
+    LOG_INFO("WeatherFlow connection established successfully.");
+    return true;
+}
+
+bool WeatherFlow::Disconnect()
+{
+    m_isConnected = false;
+    LOG_INFO("WeatherFlow disconnected.");
+    return true;
+}
+
+bool WeatherFlow::initProperties()
+{
+    INDI::Weather::initProperties();
+
+    // API Key
+    char api_key[256] = {0};
+    IUGetConfigText(getDeviceName(), "WF_API_KEY", "API_KEY", api_key, 256);
+    wfAPIKeyTP[0].fill("API_KEY", "API Key", api_key);
+    wfAPIKeyTP.fill(getDeviceName(), "WF_API_KEY", "WeatherFlow", OPTIONS_TAB, IP_RW, 60, IPS_IDLE);
```

Comment:
You can remove the first two line and simply add after the wfAPIKeyTP.fill, this line:
```
wfAPIKeyTP.load();
```

This should load the config automatically.

Reply:
Done